src/player/player-engine-dedicated-thread.ts

```diff
@@ -0,0 +1,454 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as EventEmitter from 'events';
+import * as work from 'webworkify-webpack';
+import type PlayerEngine from './player-engine';
+import Log from '../utils/logger';
+import LoggingControl from '../utils/logging-control.js';
+import { createDefaultConfig } from '../config';
+import MediaInfo from '../core/media-info';
+import MSEEvents from '../core/mse-events';
+import PlayerEvents from './player-events';
+import TransmuxingEvents from '../core/transmuxing-events';
+import SeekingHandler from './seeking-handler';
+import LoadingController from './loading-controller';
+import StartupStallJumper from './startup-stall-jumper';
+import LiveLatencyChaser from './live-latency-chaser';
+import LiveLatencySynchronizer from './live-latency-synchronizer';
+import {
+    WorkerCommandPacket,
+    WorkerCommandPacketInit,
+    WorkerCommandPacketLoggingConfig,
+    WorkerCommandPacketTimeUpdate,
+    WorkerCommandPacketReadyStateChange,
+    WorkerCommandPacketUnbufferedSeek
+} from './player-engine-worker-cmd-def.js';
+import {
+    WorkerMessagePacket,
+    WorkerMessagePacketBufferedPositionChanged,
+    WorkerMessagePacketLogcatCallback,
+    WorkerMessagePacketMSEEvent,
+    WorkerMessagePacketMSEInit,
+    WorkerMessagePacketPlayerEvent,
+    WorkerMessagePacketPlayerEventError,
+    WorkerMessagePacketPlayerEventExtraData,
+    WorkerMessagePacketTransmuxingEvent,
+    WorkerMessagePacketTransmuxingEventInfo,
+    WorkerMessagePacketTransmuxingEventRecommendSeekpoint,
+} from './player-engine-worker-msg-def.js';
+
+class PlayerEngineDedicatedThread implements PlayerEngine {
+
+    private readonly TAG: string = 'PlayerEngineDedicatedThread';
+
+    private _emitter: EventEmitter = new EventEmitter();
+    private _media_data_source: any;
+    private _config: any;
+
+    private _media_element?: HTMLMediaElement = null;
+
+    private _worker: Worker;
+    private _worker_destroying: boolean = false;
+
+    private _seeking_handler?: SeekingHandler = null;
+    private _loading_controller?: LoadingController = null;
+    private _startup_stall_jumper?: StartupStallJumper = null;
+    private _live_latency_chaser?: LiveLatencyChaser = null;
+    private _live_latency_synchronizer?: LiveLatencySynchronizer = null;
+
+    private _pending_seek_time?: number = null;
+
+    private _media_info?: MediaInfo = null;
+    private _statistics_info?: any = null;
+
+    private e?: any = null;
+
+    public static isSupported(): boolean {
+        return Worker &&
+               MediaSource &&
+               ('canConstructInDedicatedWorker' in MediaSource) &&
+               MediaSource['canConstructInDedicatedWorker'] === true;
```

Comment:
I would like to add ManagedMediaSource support in these line.
When Apple support MSE in Workers, also support  ManagedMediaSource in Workers (MMS inherits MSE)

Reply:
Since Safari is still [No signal](https://github.com/WebKit/standards-positions/issues/40) for this feature, I would like to postpone the support.